lib/server/server-middleware.js

```diff
@@ -1,234 +1,244 @@
+  middleware.push(cors({
+    origin: false,
+    methods: 'GET,HEAD,PUT,PATCH,POST,DELETE'
+  }))
+
+  // Prevent iframing of pages
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
+  middleware.push((req, res, next) => {
+    res.setHeader('X-Frame-Options', 'deny')
+    next()
+  })
+
+  // Run arbitrary routes before static routes run
+  if (options.preStaticRoutes) {
+    middleware.push(options.preStaticRoutes())
+  }
+
+  // Serve static assets
+  const routesStaticConfig = {
+    assetsUrlPrefix: ASSET_SRC_PATH,
+    staticPaths: options.staticPaths,
+    serviceSources
+  }
+  middleware.push(routesStatic.init(routesStaticConfig))
+
+  // Run arbitrary routes after static routes run
+  if (options.postStaticRoutes) {
+    middleware.push(options.postStaticRoutes())
+  }
+
+  // TODO: REMOVE IF UNNECESSARY
+  // Disable indexing of svgs
+  // https://github.com/18F/pa11y-crawl/issues/4
+  if (ENV === 'testing') {
+    options.robots = {
+      body: `User-agent: *
+disallow: /public`
+    }
+  }
+
+  // Robots
+  middleware.push(robots.init(options.robots))
+
+  // Gzip content
+  middleware.push(compression())
+
+  middleware.push(bodyParser.json())
+  middleware.push(bodyParser.urlencoded({
+    extended: false,
+    depth: 20,
+    allowDots: true,
+    parseArrays: false,
+    limit: '200kb'
+  }))
+
+  middleware.push(logger.getMiddleware())
+
+  // Use nunjucks
+  middleware.push(nunjucksConfiguration.getMiddleware())
+
+  // Local values
+  middleware.push(locals.init(ENV, options, ASSET_PATH, options.locals))
+
+  // TODO: move to editor
+  middleware.push((req, res, next) => {
+    let editmode = false
+    if (EDITABLE || !EDITABLE) {
+      const match = req._parsedUrl.pathname.match(/\/(edit|preview|flow)$/)
+      if (match) {
+        editmode = match[1]
+      }
+    }
+    Object.defineProperty(req, 'editmode', {
+      writeable: false,
+      value: editmode
+    })
+    next()
+  })
+
+  // Run everything through basic auth
+  if (options.basicAuth || USERNAME) {
+    middleware.push(auth.init(options.basicAuth || {
+      username: USERNAME,
+      password: PASSWORD,
+      realm: REALM
+    }))
+  }
+
+  // TODO: SCREENING
+  // Screen users if necessary
+  if (options.screenUsers) {
+    middleware.push(options.screenUsers(ENV))
+  }
+
+  // Fallback favicon route
+  middleware.push(['/favicon.ico', (req, res) => {
+    res.send()
+  }])
+
+  // Create user session
+  middleware.push(userSession.init({ serviceSecret: SERVICE_SECRET }))
+
+  // Keep user session alive route
+  middleware.push([KEEP_ALIVE_URL, keepAlive.init({ serviceSecret: SERVICE_SECRET })])
+
+  // Clear user session
+  middleware.push(['/reset', (req, res) => {
+    res.clearCookie('sessionId')
+    res.redirect('/')
+  }])
+
+  // Ping route
+  middleware.push([PING_URL, ping.init()])
+
+  // Healthcheck route
+  middleware.push([HEALTHCHECK_URL, healthcheck.init(options.validateHealthcheck, options)])

```

Comment:
Ditto above

Reply:
Actually, I haven't examined the circs under which "validateHealthcheck" is populated, or even whether it is